src/algebra/monoid_algebra.lean

```diff
@@ -791,6 +791,12 @@ lemma single_mul_single [has_add G] {a₁ a₂ : G} {b₁ b₂ : k} :
   (single a₁ b₁ : add_monoid_algebra k G) * single a₂ b₂ = single (a₁ + a₂) (b₁ * b₂) :=
 @monoid_algebra.single_mul_single k (multiplicative G) _ _ _ _ _ _
 
+@[simp] lemma single_pow [add_monoid G] {a : G} {b : k} :
```

Comment:
The linter times out when checking this lemma.

Reply:
I'm pretty lost here. The usual techniques to figure out what causes `simp` to loop (to which the linter links) do not give anything useful. It looks like `simp` is looping without applying any simp-lemmas.
I'll take another look later, but maybe @gebner has an idea of what is going on here.

Reply:
I think you're commenting at the wrong `single_pow` lemma.

Reply:
I believe it times out when unifying `add_monoid_algebra.semiring =?= monoid_algebra.semiring`

Reply:
Okay... that's bad news. I can't make either irreducible without breaking lots of stuff downstream. Is there some way to turn off the `simp_nf` lemma for particular lemmas? `[nolint simp_nf]` doesn't seem to help.

Reply:
I've had a hard time in the past due to not having a `single` that actually produces a term of the right type. I think this probably would go away if we added `monoid_algebra.single` as an alias for `finsupp.single`.